File: src/fixtures/bun/head.test.ts
import {
	expect,
	test,
} from "bun:test";

import {
	add,
	multiply,
	subtract,
} from "./calc";
import {
	capitalize,
	reverse,
	truncate,
} from "./strutil";

// Head tests: covers add, subtract, multiply, reverse, capitalize, truncate
// Does NOT cover: divide, modulo, padLeft

test("add", () => {
	expect(add(1, 2)).toBe(3);
	expect(add(-1, 1)).toBe(0);
});

test("subtract", () => {
	expect(subtract(5, 3)).toBe(2);
});

test("multiply", () => {
	expect(multiply(3, 4)).toBe(12);
	expect(multiply(0, 5)).toBe(0);
});

test("reverse", () => {
	expect(reverse("abc")).toBe("cba");
});

test("capitalize", () => {
	expect(capitalize("hello")).toBe("Hello");
	expect(capitalize("")).toBe("");
});

test("truncate", () => {
	expect(truncate("hello world", 5)).toBe("hello...");
	expect(truncate("hi", 10)).toBe("hi");
});

File: src/fixtures/goproject/head_test.go
package fixture

import "testing"

// Head tests: covers Add, Subtract, Multiply, Reverse, Capitalize, Truncate
// Does NOT cover: Divide, Modulo, PadLeft

func TestAdd(t *testing.T) {
	if Add(1, 2) != 3 {
		t.Fatal("expected 3")
	}
}

func TestSubtract(t *testing.T) {
	if Subtract(5, 3) != 2 {
		t.Fatal("expected 2")
	}
}

func TestMultiply(t *testing.T) {
	if Multiply(3, 4) != 12 {
		t.Fatal("expected 12")
	}
}

func TestReverse(t *testing.T) {
	if Reverse("hello") != "olleh" {
		t.Fatal("expected olleh")
	}
}

func TestCapitalize(t *testing.T) {
	if Capitalize("hello") != "Hello" {
		t.Fatal("expected Hello")
	}
}

func TestTruncate(t *testing.T) {
	if Truncate("hello world", 5) != "hello..." {
		t.Fatal("expected hello...")
	}
}
